includes/writer.js

```diff
@@ -47,6 +52,19 @@ module.exports = function(couchDbUrl, bufferSize, parallelism) {
       }
       cb();
     });
+
+    response.on('response', function(resp) {
+      if (resp.statusCode === 401) {
```

Comment:
it's only just dawned on me that we silently fail all error codes here.
did you want to deal with this in the pr? or spin out to anther issue?